pkg/opt/config.go

```diff
@@ -94,6 +99,25 @@ func (c *Config) VaultClient() (*vaultapi.Client, error) {
 	return nil, nil
 }
 
+func (c *Config) WithMetrics() (*metric.Meter, error) {
+	if c.MetricsEnabled {
+		exporter, err := prometheus.InstallNewPipeline(prometheus.Config{})
+		if err != nil {
+			return nil, err
+		}
+		http.HandleFunc("/", exporter.ServeHTTP)
+		go func() {
+			_ = http.ListenAndServe(c.MetricsAddr, nil)
+		}()
+
+		meter := exporter.MeterProvider().Meter("relay-pls")
+
+		return &meter, nil
+	}
```

Comment:
How would you feel about having a no-op exporter if metrics are disabled so we don't have to worry about nil checks down the line? The `stdout` exporter supports specifying a writer, so it looks like we could just have something like `exporter, err := stdout.InstallNewPipeline([]stdout.Option{stdout.WithWriter(ioutil.Discard)}, prometheus.Config{})`.

Reply:
Yeah, I'll look into doing that.